docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -0,0 +1,125 @@
+# Multi-Tenancy Network
+
+Xiaoning Ding, Hong Chen
+
+## Introduction
+
+This document is a part of the multi-tenancy design documents in Arktos. It describes network-related design changes to support multi-tenancy, including network model, backward-compatibility, DNS and service changes, etc.
+
+Please refer to other design documents in the "multi-tenancy" document directory if you want to learn about other parts in the multi-tenancy design.
+
+## Network Isolation
+
+Network isolation is achieved with a new API object: network:
+
+* Every pod is associated with a certain network object. If not specified it will be associated to a default network object, which is created when a space is initialized. 
+* A pod can only communicate with pods within the same network.
+* Every network has its own DNS server. The DNS server pod is automatically created when a new network object is created.
+
+
+### Network Object
+
+A network object contains its type, and type-specific configurations. For now there are two types defined: flat and vpc.
+
+A command-line parameter named "default-network-template-path" of tenant controller will decide which default network will be created for a new space.
+
+Below is the definition of a flat network:
+
+```yaml
+apiVersion: v1
```

Comment:
I'm not sure which API group and version is appropriate now. Also I'm thinking about if it's possible to define more standard network objects in a CRD manner. So that it can be used as an abstract layer.

Anyways, for now I didn't update the document. I filed an issue to track this: https://github.com/futurewei-cloud/arktos/issues/111.  Let's update the document after we have a conclusion.